app/views/thesis/process_theses.html.erb

```diff
@@ -240,15 +233,25 @@
 
     <div class="field-row layout-band layout-1q3q">
       <div class="col1q">
-        <strong>Thesis supervisor(s)*</strong>
+        <strong>Thesis supervisor(s) *</strong>
         <p class="hint">(just supervisors, not entire committee)</p>
       </div>
-      <%= f.simple_fields_for :advisors do |advisor| %>
-        <%= render 'advisor_fields', :f => advisor %>
-      <% end %>
-      <p class="links col3q">
-        <%= link_to_add_association 'Add another supervisor', f, :advisors %>
+      <% if f.object.advisors.any? %>
+        <%= f.simple_fields_for :advisors do |advisor| %>
+          <%= render 'advisor_fields', :f => advisor %>
+        <% end %>
+        <p class="links col3q">
+          <%= link_to_add_association 'Add another supervisor', f, :advisors %>
       </p>
+      <% else %>
+        <div class="advisor nested-fields col3q">
+          <div class="input field_with_hint">
+            <label class="field-label" for="advisors">Supervisor name *</label>
+            <input class="field field-text wide disabled" readonly type="text" value="There are no advisors associated with this thesis." name="advisors" id="advisors">
+            <p class="hint">See details in <%= link_to('the administrative interface.', edit_admin_thesis_path(f.object), target: :_blank) %></p>
```

Comment:
I'm not sure that I follow the logic behind not using a `link_to_add_association` link on this branch? The data model was never, IIRC, expecting that advisors would be persistent or a lookup field. The student-facing form just takes arbitrary strings and stores them, and the edit branch of this structure still has one (making it possible to add/drop supervisor strings, including those already defined).

I'm not sure what I'm missing here?

Reply:
The arbitrary string piece of this is what gave me pause. With the form field set to a readonly value, when you add an association, it retains "There are no advisors associated with this thesis" as if it were an actual value. If the processor doesn't delete that value before they save the record, then "There are no advisors associated with this thesis" is stored as an advisor. In this case, the the UI inconsistency seemed less gross than the potential data mess.

A possible alternative here is to make the readonly value a regular div that's styled like a form value. I'm not sure what implications that might have for accessibility, but it's worth a shot. I'll see what that looks like and get back to you.

Reply:
If it helps, I had to write some javascript for earlier work that hides/shows the add_association links, using some hooks that are available in the gem. I wonder how awful it would be to extend those a little more?

Reply:
Yeah, I suspect the logic would be similar for hiding/showing the placeholder text, so I can probably reuse that. Thanks for pointing that out!

Reply:
@matt-bernhardt I think I have a viable fix for this in place (accesslint struggles notwithstanding). Could you take another look?

Reply:
Yeah, this looks good to me in practice. Thanks!